inksdk/ink/src/main/java/com/microsoft/device/ink/InkView.kt

```diff
@@ -0,0 +1,277 @@
+    private var maxStrokeWidth = 10f
+
+    // properties
+    var color = Color.GRAY
+        set(value) {
+            field = value
+            currentStrokePaint.color = value
+        }
+
+    private var _dynamicPaintHandler: DynamicPaintHandler? = null
+    var dynamicPaintHandler = _dynamicPaintHandler
+        set(value) {
+            _dynamicPaintHandler = value
+            field = value
+        }
+    interface DynamicPaintHandler {
+        fun generatePaintFromPenInfo(penInfo: InputManager.PenInfo): Paint
+    }
+
+    init {
+        isOpaque = false // mae the texture view transparent!
+        // handle attributes
+        context.theme.obtainStyledAttributes(
+            attributeSet,
+            R.styleable.InkView,
+            0, 0
+        ).apply {
+            try {
+                enablePressure = getBoolean(R.styleable.InkView_enable_pressure, enablePressure)
+                color = getColor(R.styleable.InkView_ink_color, color)
+                minStrokeWidth = getFloat(R.styleable.InkView_min_stroke_width, minStrokeWidth)
+                maxStrokeWidth = getFloat(R.styleable.InkView_max_stroke_width, maxStrokeWidth)
+            } finally {
+                recycle()
+            }
+        }
+        this.surfaceTextureListener = this
+        overridePaint = Paint()
+        overridePaint.blendMode = BlendMode.SRC
+        clearPaint = Paint()
+        clearPaint.blendMode = BlendMode.CLEAR
+
+        inputManager = InputManager(
+            this,
+            object : InputManager.PenInputHandler {
+                @RequiresApi(Build.VERSION_CODES.Q)
+                override fun strokeStarted(
+                    penInfo: InputManager.PenInfo,
+                    stroke: InputManager.ExtendedStroke
+                ) {
+                    redrawTexture()
+                }
+
+                @RequiresApi(Build.VERSION_CODES.Q)
+                override fun strokeUpdated(
+                    penInfo: InputManager.PenInfo,
+                    stroke: InputManager.ExtendedStroke
+                ) {
+                    redrawTexture()
+                }
+
+                @RequiresApi(Build.VERSION_CODES.Q)
+                override fun strokeCompleted(
+                    penInfo: InputManager.PenInfo,
+                    stroke: InputManager.ExtendedStroke
+                ) {
+                    redrawTexture()
+                    strokeList += stroke
+                }
+            }
+        )
+
+        currentStrokePaint.color = color
+        currentStrokePaint.isAntiAlias = true
+        // Set stroke width based on display density.
+        currentStrokePaint.strokeWidth = TypedValue.applyDimension(
+            TypedValue.COMPLEX_UNIT_DIP,
+            minStrokeWidth,
+            resources.displayMetrics
+        )
+
+        currentStrokePaint.style = Paint.Style.STROKE
+        currentStrokePaint.strokeJoin = Paint.Join.ROUND
+        currentStrokePaint.strokeCap = Paint.Cap.ROUND
```

Comment:
Msybe it's a good idea to extract this code to a different function, something like "setupStrokePaint()"